core/src/main/java/uk/gov/justice/services/core/jms/JmsSender.java

```diff
@@ -74,16 +74,13 @@ private void send(final String queueName, final Envelope envelope) {
         try {
             final Queue queue = (Queue) getInitialContext().lookup(queueName);
 
-            try (QueueConnection queueConnection = queueConnectionFactory.createQueueConnection()) {
+            try (QueueConnection queueConnection = queueConnectionFactory.createQueueConnection();
+                 QueueSession session = queueConnection.createQueueSession(false, QueueSession.AUTO_ACKNOWLEDGE);
+                 QueueSender sender = session.createSender(queue)) {
 
-                try (QueueSession session = queueConnection.createQueueSession(false, QueueSession.AUTO_ACKNOWLEDGE)) {
+                sender.send(envelopeConverter.toMessage(envelope, session));
```

Comment:
If you only have one try with resource will the other resources be automatically closed on exceptions?